presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkSettingsRequirements.java

```diff
@@ -49,6 +50,7 @@ public void verify(SparkContext sparkContext, Session session)
                 "grouped execution is not supported");
         verify(!isRedistributeWrites(session), "redistribute writes is not supported");
         verify(!isScaleWriters(session), "scale writes is not supported");
+        verify(!isPreferDistributedUnion(session), "prefer distributed union is expected to be disabled");
```

Comment:
I am actually thinking to set `PREFER_DISTRIBUTED_UNION ` to be false in production -- this would fail user query loudly instead of make it run but somehow very slow...

`PREFER_DISTRIBUTED_UNION` should only be used to entertain test.

Reply:
That makes sense. Let me change that.